pkg/controllers/job/job_controller_actions.go

```diff
@@ -265,10 +265,6 @@ func (cc *Controller) syncJob(jobInfo *apis.JobInfo, updateStatus state.UpdateSt
 					pod.Name, job.Name, err)
 				creationErrs = append(creationErrs, fmt.Errorf("failed to create pod %s, err: %#v", pod.Name, err))
 			} else {
-				if err != nil && apierrors.IsAlreadyExists(err) {
```

Comment:
Need provide more detail why we delete this resync logic here rather than fix the "can not find pod <namespace/pod> in cache" issue.

Reply:
I think there is a race between the core job controller and the resync controller.

And same race between `JobInfo.AddPod` and `controller.syncJob`

Reply:
> I think there is a race between the core job controller and the resync controller.

Please share your analysis of this issue firstly before PR.

Reply:
I will add the analysis to Pr description